docs/users/topics/phpstorm.md

```diff
@@ -10,7 +10,7 @@ This is based on the wonderful [article](https://susi.dev/fully-integrate-ddev-a
 
 - PhpStorm 2021.2 or higher
 - You may need to add the `PHP Docker` plugin.
-- docker-compose v1. Due to a [bug or two in PhpStorm](https://youtrack.jetbrains.com/issue/WI-61205), it can't properly interpret docker-compose v2 output as of Sept 2021.  If this is still a problem, you can `ddev config global --required-docker-compose-version=v1.29.2` to globally require docker-compose v1.
+- As of version v1.18.2, DDEV is bundled with its own docker-compose binary, using Compose V2 by default. Enable _Use Compose V2_ in PHPStorm _Settings/Preferences | Build, Execution, Deployment | Docker | Tools_, in case you have an up to date docker-compose binary installed (Check with `docker-compose version` > 2.0). Otherwise, you can also set the bundled docker-compose binary as _Docker Compose executable_ which you can be found in `~/.ddev/bin`. In this case, you do not need to select the _Use Compose V2_ Checkbox.
```

Comment:
Thanks so much for chasing this and this contribution!

I would say we don't need to make it this complex. That's a lot of information.

```suggestion
- DDEV uses docker-compose v2 by default, so enable the **Use Compose V2** checkbox  in `PHPStorm _Settings/Preferences | Build, Execution, Deployment | Docker | Tools`.
```

Reply:
I am not sure if we should ommit all of the information. You are right, that it might be to detailed. But Compose V2 is currently only avaiable by default on Systems with Docker Desktop (Windows, Mac). I think Linux Users might not have Compose V2 support of their system binary, so we should give information on how to use it anyway, or at least how to find out if they have.
What do you think?

Reply:
No, all ddev users on all platforms have docker-compose v2 unless they override globally.

Reply:
Eanbling the Use Compose V2 Checkbox will let PHPStorm use the docker-compose executable of your system to parse the DDEV generated configuration. Selecting the Checkbox will not guarantee a V2 capable binary will be used. As far as I understood it will just change its internal handling of the binary and validation.
If you like I can try to confirm this on a Debian distribution.

Reply:
I recommend telling people to set the docker-compose binary to ~/.ddev/bin/docker-compose I guess. That's a consistent solution. People who need to do something more exotic can sort it out, but since PhpStorm is now working with docker-compose v2, they shouldn't need anything more exotic.

Reply:
This is a too restricted view in my opinion. I work on many projects every day and only for a few I use DDEV. Setting the docker-compose binary is a global setting affecting all Projects. This might result in unexpected behavior for any non ddev project. It is a valid solution for people using docker-compose only in DDEV context, that's why I listed it as an option.
I would recomment using the system docker-compose binary, as it is in sync with your installed docker version (As V2 is a docker plugin) and it is the same in both your OS and WSL (Thanks to the WSL2 Backend of Docker Desktop). Also it is automatically updated by Docker Desktop or the package manager of your choice. 
Another less idealistic reason for my recommendation is, that I work mostly in WSL. It is currently not possible to select a Linux executable (non .exe) in PHPStorm as docker-compose executable. So you would need to install DDEV in WSL, as this is currently the recommended way, and in windows to select its binary here. 
This makes in total 3 docker-compose binaries you have which might all have different versions. This will cause trouble for sure.

Reply:
Sorry, these are the ddev instructions, so let's use the ddev docker-compose. People like you will know what to do, and people who have need for another docker-compose will know what to do. 

It's always complicated to understand which PhpStorm instructions are global for all projects and which are per-project, so that may be confusing, but again, we're doing the ddev instructions here.

Reply:
Okay, so let's summarize:
- The original version works in all environments but might be too complicated, I admit that.
- The version that is currently suggested by you won't work for Linux users in most cases (No V2 binary)
- Your last suggestion was using the bundled version, this will not work for Windows with WSL (Can't select Linux binaries)

We could close this pull request, as downgrading DDEV should work on all environments. You should have permission to push to this pull request, so you can choose whatever version you like most.

Reply:
Thanks so much for your work on this and for pioneering all this and testing it. 

If you're using PhpStorm on the Windows side (and Docker Desktop) with WSL2, yes, you can't point it directly at ddev's binary.  However, the bundled Docker Desktop version will work, and they all default to v2. Thanks for pointing that out. WSL2 always brings out the complexity. Or at least WSL2 with Docker Desktop. 

How about this?

```suggestion
DDEV uses docker-compose v2 by default, so enable the **Use Compose V2** checkbox  in `PHPStorm _Settings/Preferences | Build, Execution, Deployment | Docker | Tools` and set the "Docker Compose executable" to ddev's private docker-compose binary at `~/.ddev/bin/docker-compose` (Home directory, `.ddev/bin/docker-compose.). However, if using PhpStorm on Windows side with WSL2 you'll need to use a docker-compose v2 binary as provided by Docker Desktop, normally `C:\Program Files\Docker\Docker\resources\bin\docker-compose`.
```

Reply:
This one seems to cover all use cases that come to my mind. I accepted your suggestion.